sunpy/coordinates/screens.py

```diff
@@ -120,6 +196,12 @@ def calculate_distance(self, frame):
         # Ignore sqrt of NaNs
         with np.errstate(invalid='ignore'):
             distance = ((-1*b) + np.sqrt(b**2 - 4*c)) / 2  # use the "far" solution
+
+        # Iterate the calculation if differential rotation is being applied
+        if _transformations._autoapply_diffrot is not None and frame.obstime != self._center.obstime:
+            screen_frame = Helioprojective(observer=self._center, obstime=self._center.obstime)
+            distance = self._iterate_calculate_distance(frame, distance, screen_frame)
+
```

Comment:
it would be nice if this didn't have to be repeated for every screen subclass, but for two subclasses it's not that big of a deal.